pkg/resourceutil/sort_test.go

```diff
@@ -17,31 +17,88 @@ package resourceutil
 import (
 	"testing"
 
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/assert"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 func TestSortResourcesByKind(t *testing.T) {
-	resources := []Resource{
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Secret"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRole"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "IngressRoute"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRoleBinding"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "PodSecurityPolicy"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "ServiceAccount"}},
-		{GroupVersionKind: &schema.GroupVersionKind{Kind: "Service"}},
-	}
-
 	t.Run("Reordering resources based on default reordering function", func(t *testing.T) {
-		expected := []string{"PodSecurityPolicy", "ServiceAccount", "ClusterRole", "ClusterRoleBinding", "Secret", "ConfigMap", "ConfigMap", "Service", "Deployment", "IngressRoute", "Unknown"}
+		resources := []Resource{
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Unknown"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Secret"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRole"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "IngressRoute"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ClusterRoleBinding"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ConfigMap"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Deployment"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "PodSecurityPolicy"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "ServiceAccount"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "SecretProviderClass"}},
+			{GroupVersionKind: &schema.GroupVersionKind{Kind: "Service"}},
+		}
+		expected := []string{
+			"PodSecurityPolicy",
+			"ServiceAccount",
+			"ClusterRole",
+			"ClusterRoleBinding",
+			"SecretProviderClass",
+			"Secret",
+			"ConfigMap",
+			"ConfigMap",
+			"Service",
+			"Deployment",
+			"IngressRoute",
+			"Unknown",
+		}
 		var orderedNames []string
 		for _, resource := range SortResourcesByKind(resources, nil) {
 			orderedNames = append(orderedNames, resource.GroupVersionKind.Kind)
 		}
-		require.ElementsMatch(t, expected, orderedNames)
+
+		assert.Exactly(t, expected, orderedNames)
```

Comment:
It's better to use require instead of assert. assert returns a bool, instead require will end the test if fails the validation (it calls t.FailNow())

Reply:
In this context they act the same since the assertion is the last instruction in the test. However is more conceptually correct to use `require`. Im gonna change it :+1: